addons/spreadsheet_dashboard/static/tests/dashboard/dashboard_action.test.js

```diff
@@ -328,3 +349,84 @@ test("Changing filter values will create a new share", async function () {
         `localhost:8069/share/url/2`
     );
 });
+
+test("Cog menu displays 'Edit' and 'Delete' options when clicked", async function () {
+    await createSpreadsheetDashboard();
+
+    const cogIcon = await openCogMenu();
+    expect(cogIcon).toHaveClass("show");
+    expect(getFixture().querySelectorAll(".o_popover")).toHaveLength(1);
+
+    const popover = getFixture().querySelector(".o_popover");
+    const dropdownItems = popover.querySelectorAll(".o-dropdown-item");
+
+    expect(dropdownItems).toHaveLength(2);
+    const [editOption, deleteOption] = dropdownItems;
+
+    expect(editOption.innerText).toBe("Edit");
+    expect(editOption.querySelector("i")).toHaveClass("fa-pencil");
+
+    expect(deleteOption.innerText).toBe("Delete");
+    expect(deleteOption.querySelector("i")).toHaveClass("fa-trash");
+});
+
+test("Clicking 'Edit' option navigates to dashboard edit view", async function () {
+    const action = {
+        type: "ir.actions.client",
+        tag: "action_edit_dashboard",
+        params: {
+            spreadsheet_id: 1,
+        },
+    };
+
+    await createSpreadsheetDashboard({
+        mockRPC: async function (route, args) {
+            if (args.method === "action_edit_dashboard" && args.model === "spreadsheet.dashboard") {
+                expect.step("action_edit_dashboard");
+                return action;
+            }
+        },
+    });
+
+    await openCogMenu();
+    await clickEditOption();
+    const env = getMockEnv();
+    patchWithCleanup(env.services.action, {
```

Comment:
Not 100% sure how it works, you first click the option then you patch the action service to observe what happens. The order should be inversed